repos/system_upgrade/el7toel8/actors/localreposinhibit/actor.py

```diff
@@ -0,0 +1,86 @@
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.models import TargetRepositories, UsedTargetRepositories
+from leapp.reporting import Report
+from leapp.tags import IPUWorkflowTag, TargetTransactionChecksPhaseTag
+
+
+class LocalReposInhibit(Actor):
+    """Inhibits the upgrade if local repositories were found.
+
+    Currently we're not supporting local repositories during the upgrade.
+    There is a workaround available, which described inside the remediation of
+    the report.
+
+    We're looking for a baseurl only inside TargetRepositories.custom_repos
+    because local repos might exists as only custom repository, due to the
+    fact that they might be specified only inside custom repo file of the leapp
+    or via --enablerepo flag. Both options puts the repo into
+    CustomTargetRepository.
+    """
+
+    name = "local_repos_inhibit"
+    consumes = (UsedTargetRepositories, TargetRepositories)
+    produces = (Report,)
+    tags = (IPUWorkflowTag, TargetTransactionChecksPhaseTag)
+
+    def get_used_custom_repos(self):
+        """Get used custom repositories.
+
+        :returns: Generator[CustomTargetRepository]
+        """
+        # fmt: off
+        used_target_repos = next(self.consume(UsedTargetRepositories)).repos
+        custom_target_repos = next(self.consume(TargetRepositories)).custom_repos
+        # fmt: on
+        used_target_repos_ids = [
+            used_tg_repo.repoid for used_tg_repo in used_target_repos
+        ]
+        return filter(  # pylint: disable-msg=deprecated-lambda, filter-builtin-not-iterating
+            lambda custom_target_repo: custom_target_repo.repoid
+            in used_target_repos_ids
+            and custom_target_repo.baseurl,
+            custom_target_repos,
+        )
+
+    def process(self):
+        for custom_repo in self.get_used_custom_repos():
+            if custom_repo.baseurl.startswith("file:"):
```

Comment:
it should be checked against the real info on the system. repositories enabled using `--enablerepo` do not have set the baseurl in custom repos (for various valid reasons; e.g. the repository is not defined in the repodir before run of the leapp - e.g. when the repofile is generated/copied/... into the /etc/yum.repos.d dir dynamically during the run of the leapp; e.g if you want to enable specific RH repository that is not mapped by our repomap file, ...).